src/main/java/com/github/manolo8/darkbot/core/manager/GuiManager.java

```diff
@@ -72,28 +79,43 @@
 
     private boolean needRefresh;
 
-    public GuiManager(Main main) {
+    public GuiManager(Main main, PluginAPI pluginAPI) {
         this.main = main;
+        this.pluginAPI = pluginAPI;
+        this.slotBarsProxy = pluginAPI.requireInstance(SlotBarsProxy.class);
+        this.settingsProxy = pluginAPI.requireInstance(SettingsProxy.class);
 
         this.validTime = System.currentTimeMillis();
 
-        this.pet = register("pet", new PetManager(main));
-        this.oreTrade = register("ore_trade", new OreTradeGui(main));
-        this.group = register("group", new GroupManager(main));
+        this.pet = register("pet", PetManager.class);
+        this.oreTrade = register("ore_trade", OreTradeGui.class);
+        this.group = register("group", GroupManager.class);
 
         this.main.status.add(value -> validTime = System.currentTimeMillis());
+        this.lostConnection = register("lost_connection");
+        this.connecting = register("connection");
+        this.quests = register("quests");
+        this.minimap = register("minimap");
+        this.targetedOffers = register("targetedOffers", TargetedOfferGui.class);
+        this.logout = register("logout");
+        this.eventProgress = register("eventProgress");
+        this.eternalGate = register("eternal_gate");
+        this.blacklightGate = register("eternal_blacklight");
+        this.refinement = register("refinement", RefinementGui.class);
     }
 
     public Gui register(String key) {
-        return register(key, new Gui());
+        return register(key, Gui.class);
     }
 
-    public <T extends Gui> T register(String key, T gui) {
-        @SuppressWarnings("UnnecessaryLocalVariable")
-        Gui guiFix = gui; // Workaround for a java compiler assertion bug having issues with types
+    @SuppressWarnings({"unchecked", "CastCanBeRemovedNarrowingVariableType"})
+    public <T extends Gui> T register(String key, Class<T> gui) {
+        Gui guiFix = pluginAPI.requireInstance(gui);
         this.guis.addLazy(key, guiFix::update);
         this.registeredGuis.add(guiFix);
-        return gui;
+
+        // Workaround for a java compiler assertion bug having issues with types
```

Comment:
The compiler bug is not in the last line, it's about `gui::update` not working with a generic type, so we use `guiFix` that is "hard-coded" to be an instance of a Gui class just so that java compiler doesn't blow up. I don't think the comment should be on the last line.